WordPress/openverse
# Enabling fetching sensitive results when the UI switch is toggled

## Fixes



Fixes #2879 by @obulat

## Description


This PR enables the users to fetch the sensitive results by toggling the UI switch in the filters sidebar/modal.

To do so, a new `setFilter` method is added to the search store. When the switch is toggled, `setFilter` is called to update the value in the store to match the UI value. This enables users to fetch `sensitive` results.

Note that as per the Implementation plan requirements, the URL on the frontend does not have the parameter for sensitivity set, but the API requests still set the `unstable__include_sensitive_results` parameter set to true if the toggle is On.

Not directly related to this issue, but this PR moves the watcher that triggers media fetching on changes to the filters from `VSearchGridFilter` to `search.vue` (which already handles fetching on changes to the query parameters)

## Testing Instructions

I've updated the PR in the last commit to use the `unstable__sensitivity` values as the Media.sensitivity values, so now sensitive toggle works as expected, and you can actually see the results blurring based on the API filtered index values.

Enable the `sensitive_results` feature flag on the `/preferences` page.
Go back to the homepage and search for "cat". Notice that there are no query parameters in the URL besides `q=cat`.
Go to the filters sidebar (or the filters tab in the content switcher modal on a mobile screen), and turn on the "Sensitive results" switch. Notice that the URL ~~is changed to http://localhost:8443/search/?q=cat&unstable__include_sensitive_results=true~~ **doesn't change**, and a new set of results is loaded.

If you reload the same page (using CMD+R or the browser's Reload button), it should not show sensitive results (blurred, if you didn't unblur them using the switch) because we don't have a cookie for `fetch_sensitive` (only for `sensitive content`). Should we actually add this cookie?

If you try searching for something that is text-sensitive, you should see mostly blurred results.
When you turn off the "Sensitive results" switch, you should see no blurred media items in the results. 

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
